### ISSUE TITLE
Files are listed in prompt !?

### ISSUE THREAD

**Issue by chmike** [magicmonty/bash-git-prompt#295]
I have the latest version of bash-git-prompt, but I suddenly see the files in the local directory listed in the prompt. 

Here is how it looks like. Note that I have customized the prompt. But I didn't have this problem until today. I didn't modify the customization today. The problem shows up with a fresh new shell.

```
xxx@yyy:~$ cd Sources/
xxx@yyy:~/Sources$ mkdir testPrompt
xxx@yyy:~/Sources$ cd testPrompt/
xxx@yyy:~/Sources/testPrompt$ git init
Dépôt Git vide initialisé dans /home/xxx/Sources/testPrompt/.git/
- ~/Sources/testPrompt [master *|-] 
15:54 $ echo "This is dummy text" > file.txt
- ~/Sources/testPrompt [master file.txt|…1] 
15:55 $ git add file.txt 
- ~/Sources/testPrompt [master file.txt|^1] 
15:55 $ git commit -m "test"
[master (commit racine) 1a98537] test
 1 file changed, 1 insertion(+)
 create mode 100644 file.txt
- ~/Sources/testPrompt [master file.txt|-] 
15:56 $ git checkout -b test
Basculement sur la nouvelle branche 'test'
- ~/Sources/testPrompt [test file.txt|-] 
15:56 $ echo "This is dummy text 2" > file2.txt
- ~/Sources/testPrompt [test file2.txt file.txt|…1] 
15:56 $ 
```

The weird thing is that the problem doesn't show up with git repositories untouched today. 
I have the latest release of bash-git-prompt. 

This is my customized theme. As you can see I don't touch the prompt itself.

```
override_git_prompt_colors() {
  GIT_PROMPT_THEME_NAME="ChM2 Solarized Extravagant"

  GIT_PROMPT_BRANCH="${Green}"
  GIT_PROMPT_REMOTE=" "
  GIT_PROMPT_SYMBOLS_NO_REMOTE_TRACKING="*"

  GIT_PROMPT_SEPARATOR="|"
  GIT_PROMPT_CHANGED="${BoldCyan}!"
  GIT_PROMPT_STAGED="${Yellow}^"
  GIT_PROMPT_UNTRACKED="${Blue}…"
  GIT_PROMPT_CONFLICTS="${BoldRed}?"
  GIT_PROMPT_STASHED="${Magenta}>"
  GIT_PROMPT_CLEAN="${BoldGreen}-"
  GIT_PROMPT_COMMAND_OK="${BoldGreen}-"
  GIT_PROMPT_COMMAND_FAIL="${BoldRed}X"
}

reload_git_prompt_colors "ChM2 Solarized Extravagant"
```

Note that I use gitless, but I have reproduced the problem, as shown above, with a new repository never touched with gitless.

**Comment by ogr3** [comment]
What OS and bash-version are you using?

What is the output of running 

`your/bash-git-prompt/installation/gitstatus.sh`

in the problematic repo?

**Comment by chmike** [comment]
I'm using Ubuntu 16.04. 

This is the output

```
X ~/Sources/testPrompt [test file2.txt file.txt|…1] 
19:50 $ ~/.bash-git-prompt/gitstatus.sh
test
_NO_REMOTE_TRACKING_
^
0
0
0
1
0
0
```

**Comment by chmike** [comment]
After a few debug tests I have found out that the list of files result from using `*` as the GIT_PROMPT_SYMBOLS_NO_REMOTE_TRACKING symbol.

When I replace this symbol with `0` the problem disappear. The problem/bug may have been introduced in latest patches. I only saw it when I had a branch without up-stream. 

The only location where GIT_PROMPT_SYMBOLS_NO_REMOTE_TRACKING is used is in the replaceSymbols function in the gitprompt.sh script. 

It's strange that `*` is replaced by the list of local files. 

Defining GIT_PROMPT_SYMBOLS_NO_REMOTE_TRACKING as `\*` didn't work. `\*` is displayed. 
I changed it into `${Green}-`.

**Comment by magicmonty** [comment]
ok, I have a look in it. But currently I'm not sure how to fix it.
It seems that `*` has a special meaning in ZSH, so if i call `echo *`, I get the file list, as seen in the prompt.

**Comment by chmike** [comment]
Aha. I don't know how we can avoid that. That is why I tried the `\*` but it didn't work. I got around it by changing the symbol. 

Here is what I found to avoid the globa expansion http://stackoverflow.com/questions/11456403/stop-shell-wildcard-character-expansion/11456496#11456496

**Comment by magicmonty** [comment]
Thanks, `set -f` seems to do the trick

**Comment by magicmonty** [comment]
Can you try this fix please?

**Comment by chmike** [comment]
I tried with the latest commit 
```
commit 6645a82003d552f7411f574499b26cc1961838f2
Author: Martin Gondermann <magicmonty@pagansoft.de>
Date:   Sat Nov 5 21:44:35 2016 +0100

    Disable globbing on replacing prompt symbols (fixes #295)
```
and it works. It is now possible to use a `*` as following in a theme. 

```
GIT_PROMPT_SYMBOLS_NO_REMOTE_TRACKING="*"
```
